gnomad_qc/v5/sample_qc/identify_trios.py

```diff
@@ -0,0 +1,575 @@
+    fake_ped = create_fake_pedigree(
+        n=n_fake_trios,
+        sample_list=list(
+            filter_ht.filter(filter_ht.outlier_filtered, keep=False).s.collect()
+        ),
+        real_pedigree=ped,
+    )
+    return fake_ped
+
+
+def run_mendel_errors(
+    vds: hl.vds.VariantDataset,
+    ped: hl.Pedigree,
+    fake_ped: hl.Pedigree,
+) -> hl.Table:
+    """
+    Run Hail's `mendel_errors` on chr20 of the VDS subset to samples in `ped` and `fake_ped`.
+
+    :param vds: Input VariantDataset.
+    :param ped: Inferred Pedigree.
+    :param fake_ped: Fake Pedigree.
+    :return: Table with Mendel errors on chr20.
+    """
+    merged_ped = hl.Pedigree(trios=ped.trios + fake_ped.trios)
+    ped_samples = [s for t in merged_ped.trios for s in [t.s, t.pat_id, t.mat_id]]
+
+    logger.info("Sub-setting VDS to %i samples...", len(ped_samples))
+    vds = hl.vds.filter_samples(vds, ped_samples)
+    vds.variant_data = vds.variant_data.select_entries("LA", "LGT")
+    mt = hl.vds.to_dense_mt(vds)
+
+    mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
+
+    logger.info("Running Mendel errors for %s trios.", len(merged_ped.trios))
+    mendel_err_ht, _, _, _ = hl.mendel_errors(mt["GT"], merged_ped)
+
+    return mendel_err_ht
+
+
+def filter_ped(
+    ped: hl.Pedigree,
+    mendel_ht: hl.Table,
+    max_mendel_z: Optional[int] = 3,
+    max_de_novo_z: Optional[int] = 3,
+    max_mendel_n: Optional[int] = None,
+    max_de_novo_n: Optional[int] = None,
+) -> Tuple[hl.Pedigree, Dict[str, Dict[str, int]]]:
+    """
+    Filter a Pedigree based on Mendel errors and de novo metrics.
+
+    :param ped: Pedigree to filter.
+    :param mendel_ht: Table with Mendel errors.
+    :param max_mendel_z: Optional maximum z-score for Mendel error metrics. Default is 3.
+    :param max_de_novo_z: Optional maximum z-score for de novo metrics. Default is 3.
+    :param max_mendel_n: Optional maximum Mendel error count. Default is None.
+    :param max_de_novo_n: Optional maximum de novo count. Default is None.
+    :return: Tuple of filtered Pedigree and dictionary of filtering parameters.
+    """
+    cutoffs = {
+        "mendel": (
+            max_mendel_z,
+            max_mendel_n,
+        ),
+        "de_novo": (
+            max_de_novo_z,
+            max_de_novo_n,
+        ),
+    }
+
+    # Check that only one of `max_mendel_z` or `max_mendel_n` and one of `max_de_novo_z`
+    # or `max_de_novo_n` are set. If both are set, favor using the max number over max
+    # std dev.
+    cutoffs_by_method = defaultdict(dict)
+    for m, (max_z, max_n) in cutoffs.items():
+        if max_n:
+            if max_z:
+                logger.warning(
+                    "Both `max_%s_z` and `max_%s_n` are set. Using `max_%s_n` of %d!",
+                    *(m,) * 3,
+                    max_n,
+                )
+            cutoffs_by_method["count"][m] = max_n
+        elif max_z:
+            cutoffs_by_method["stdev"][m] = max_z
+
```

Comment:
Since everything is optional, lets just add a else: logger.warning("No max cutoff passed. Please pass at least max count or st dev.") Or something